src/Evolution/Particles/MonteCarlo/Actions/InitializeMonteCarlo.hpp

```diff
@@ -0,0 +1,167 @@
+#include "Evolution/Particles/MonteCarlo/Packet.hpp"
+#include "Evolution/Particles/MonteCarlo/Tags.hpp"
+#include "Parallel/AlgorithmExecution.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MutateAssign.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "PointwiseFunctions/InitialDataUtilities/InitialData.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace evolution::initial_data::Tags {
+struct InitialData;
+}  // namespace evolution::initial_data::Tags
+
+namespace tuples {
+template <typename...>
+class TaggedTuple;
+}  // namespace tuples
+
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace Initialization::Actions {
+
+/// \ingroup InitializationGroup
+/// \brief Allocate variables needed for evolution of Monte Carlo transport
+///
+/// Uses:
+/// - evolution::dg::subcell::Tags::Mesh<dim>
+/// - evolution::dg::subcell::Tags::Coordinates<dim, Frame::Inertial>
+/// - evolution::initial_data::Tags::InitialData
+///
+/// DataBox changes:
+/// - Adds:
+///   * Particles::MonteCarlo::Tags::PacketsOnElement
+///   * Particles::MonteCarlo::Tags::RandomNumberGenerator
+///   * Particles::MonteCarlo::Tags::DesiredPacketEnergyAtEmission<
+///                                  NeutrinoSpecies>
+///   * Particles::MonteCarlo::Tags::CellLightCrossingTime<DataVector>
+///   * Background hydro variables
+///   * Particles::MonteCarlo::Tags::MortarDataTag<dim>
+///   * Particles::MonteCarlo::Tags::McGhostZoneDataTag<dim>
+///
+/// - Removes: nothing
+/// - Modifies: nothing
+template <typename System, size_t EnergyBins, size_t NeutrinoSpecies>
+struct InitializeMCTags {
+ public:
+  using hydro_variables_tag = typename System::hydro_variables_tag;
+
+  static constexpr size_t dim = System::volume_dim;
+  using simple_tags =
+      tmpl::list<Particles::MonteCarlo::Tags::PacketsOnElement,
+                 Particles::MonteCarlo::Tags::RandomNumberGenerator,
+                 Particles::MonteCarlo::Tags::DesiredPacketEnergyAtEmission<
+                     NeutrinoSpecies>,
+                 Particles::MonteCarlo::Tags::CellLightCrossingTime<DataVector>,
+                 hydro_variables_tag,
+                 Particles::MonteCarlo::Tags::MortarDataTag<dim>,
+                 Particles::MonteCarlo::Tags::McGhostZoneDataTag<dim>>;
+
+  using compute_tags = tmpl::list<>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static Parallel::iterable_action_return_t apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) {
+    const size_t num_grid_points =
```

Comment:
This assumes that Tags::ActiveGrid is ActiveGrid::Subcell.  Instead check what the value is and either ERROR or update to handle correctly if it is ActiveGrid::Dg